plugins/modules/aci_l3out_extepg_to_route_control_profile.py

```diff
@@ -0,0 +1,298 @@
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        l3out=dict(type="str", aliases=["l3out_name"]),
+        ext_epg=dict(type="str", aliases=["external_epg"]),
+        profile=dict(type="str", aliases=["profile_name", "route_control_profile"]),
+        direction=dict(type="str", aliases=["route_control_profile_direction"], choices=["import", "export"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["l3out", "tenant", "profile", "ext_epg", "direction"]],
+            ["state", "present", ["l3out", "tenant", "profile", "ext_epg", "direction"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    l3out = module.params.get("l3out")
+    state = module.params.get("state")
+    tenant = module.params.get("tenant")
+    profile = module.params.get("profile")
+    direction = module.params.get("direction")
+    ext_epg = module.params.get("ext_epg")
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="l3extOut",
+            aci_rn="out-{0}".format(l3out),
+            module_object=l3out,
+            target_filter={"name": l3out},
+        ),
+        subclass_2=dict(
+            aci_class="l3extInstP",
+            aci_rn="instP-{0}".format(ext_epg),
+            module_object=ext_epg,
+            target_filter={"name": ext_epg},
+        ),
+        subclass_3=dict(
+            aci_class="l3extRsInstPToProfile",
+            aci_rn="rsinstPToProfile-[{0}]-{1}".format(profile, direction),
+            module_object=direction,
+            target_filter={"direction": direction},
```

Comment:
should we add filter `"tnRtctrlProfileName": profile` to target filter?

Reply:
added